Limit features in one tile

I tried to slice the input `QuerySet` like this: `self.queryset = input_queryset[:10000]` to limit features per tile. I got this error:

```
Fail: AssertionError: Cannot filter a query once a slice has been taken.
```

Thinking about it, it does not make sense to slice before the spatial intersection, of course. So this would have to be done after this line:

https://github.com/submarcos/django-vectortiles/blob/fb2ad8ca8ea9c4384a5325599c95f363ce9e89f9/vectortiles/postgis/mixins.py#L20

FYI: [`pg_tileserv`](https://github.com/CrunchyData/pg_tileserv) has a [default features per tile of 10000](https://github.com/CrunchyData/pg_tileserv#configuration-file).

--- a/test_vectortiles/test_app/views.py
+++ b/test_vectortiles/test_app/views.py
@@ -15,6 +15,7 @@ class MapboxFeatureView(MapboxMVTView, ListView):
     model = Feature
     vector_tile_layer_name = "features"
     vector_tile_fields = ('name', )
+    vector_tile_queryset_limit = 100
 
 
 class MapboxLayerView(MapboxMVTView, DetailView):
@@ -36,6 +37,7 @@ class PostGISFeatureView(PostgisMVTView, ListView):
     model = Feature
     vector_tile_layer_name = "features"
     vector_tile_fields = ('name', )
+    vector_tile_queryset_limit = 100
 
 
 class PostGISFeatureViewWithManualVectorTileQuerySet(PostgisMVTView, DetailView):

--- a/vectortiles/mapbox/mixins.py
+++ b/vectortiles/mapbox/mixins.py
@@ -29,6 +29,10 @@ class MapboxBaseVectorTile(BaseVectorTileMixin):
             f"{self.vector_tile_geom_name}__intersects": bbox
         }
         features = features.filter(**filters)
+        # limit feature number if limit provided
+        limit = self.get_vector_tile_queryset_limit()
+        if limit:
+            features = features[:limit]
         features = features.annotate(clipped=Intersection(Transform(self.vector_tile_geom_name, 3857), bbox))
         if features:
             tile = {

--- a/vectortiles/mixins.py
+++ b/vectortiles/mixins.py
@@ -7,6 +7,7 @@ from django.http import HttpResponse
 class BaseVectorTileMixin:
     vector_tile_content_type = "application/x-protobuf"
     vector_tile_queryset = None
+    vector_tile_queryset_limit = None
     vector_tile_layer_name = None  # name for data layer in vector tile
     vector_tile_geom_name = "geom"  # geom field to consider in qs
     vector_tile_fields = None  # other fields to include from qs
@@ -20,6 +21,9 @@ class BaseVectorTileMixin:
 
     def get_vector_tile_queryset(self):
         return self.vector_tile_queryset if self.vector_tile_queryset is not None else self.get_queryset()
+
+    def get_vector_tile_queryset_limit(self):
+        return self.vector_tile_queryset_limit
 
     def get_vector_tile_layer_name(self):
         return self.vector_tile_layer_name

--- a/vectortiles/postgis/mixins.py
+++ b/vectortiles/postgis/mixins.py
@@ -25,6 +25,10 @@ class PostgisBaseVectorTile(BaseVectorTileMixin):
                                                              buffer,
                                                              clip_geom))
         fields = self.vector_tile_fields + ("geom_prepared", ) if self.vector_tile_fields else ("geom_prepared", )
+        # limit feature number if limit provided
+        limit = self.get_vector_tile_queryset_limit()
+        if limit:
+            features = features[:limit]
         # keep values to include in tile (extra included_fields + geometry)
         features = features.values(*fields)
         # generate MVT

